shared_model/backend/protobuf/query_responses/proto_query_response.hpp

```diff
@@ -0,0 +1,96 @@
+/**
+ * Copyright Soramitsu Co., Ltd. 2017 All Rights Reserved.
+ * http://soramitsu.co.jp
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef IROHA_PROTO_QUERY_RESPONSE_HPP
+#define IROHA_PROTO_QUERY_RESPONSE_HPP
+
+#include "backend/protobuf/query_responses/proto_account_asset_response.hpp"
+#include "interfaces/queries/query.hpp"
+#include "interfaces/query_responses/query_response.hpp"
+#include "responses.pb.h"
+#include "utils/lazy_initializer.hpp"
+#include "utils/variant_deserializer.hpp"
+
+template <typename... T, typename Archive>
+auto loadQueryResponse(Archive &&ar) {
+  int which =
+      ar.GetDescriptor()->FindFieldByNumber(ar.response_case())->index();
+  return shared_model::detail::variant_impl<T...>::template load<
+      shared_model::interface::QueryResponse::QueryResponseVariantType>(
```

Comment:
Need to add check of the inheritance here

Reply:
What do you mean?